[refactor] - remove redundant chunking

ahrav opened the pull request:
<!--
Please create an issue to collect feedback prior to feature additions. Please also reference that issue in any PRs.
If possible try to keep PRs scoped to one feature, and add tests for new features.
-->

### Description:
This PR removes redundant chunking from the engine. All chunks supplied to the ChunksChan are already sized appropriately.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

ahrav committed:
remove redundant chunking

rosecodym reviewed (approved):
Why were we doing this before?

ahrav commented:
> Why were we doing this before?



> Why were we doing this before?

Originally, sources would place their data into a chunk directly. Later, we changed this to use our `chunkReader`, which chunks the data before creating a chunk. However, we ended up with chunking in both places, rendering the chunking in the engine ineffective. This redundancy wasted cycles, created and destroyed goroutines, and added an unnecessary level of indentation 🤣.
